src/lib.rs

```diff
@@ -260,23 +308,74 @@ impl Iterator for Paths {
 
 }
 
-fn list_dir_sorted(path: &Path) -> Option<Vec<Path>> {
-    match fs::readdir(path) {
-        Ok(mut children) => {
-            children.sort_by(|p1, p2| p2.filename().cmp(&p1.filename()));
-            Some(children.into_iter().collect())
-        }
-        Err(..) => None
-    }
+/// A pattern parsing error.
+#[define(Copy)]
```

Comment:
It's ok to avoid this for now as avoiding `Copy` will allow us to expand in the future to perhaps a linear representation.